internal/datasourcev2/streaming_data_source.go

```diff
@@ -287,6 +287,7 @@ func (sp *StreamProcessor) consumeStream(stream *es.Stream, closeWhenReady chan<
 				sp.setInitializedAndNotifyClient(true, closeWhenReady)
 
 			default:
+				processedEvent = false
```

Comment:
This seems incorrect in both the fdv1 and fdv2 sources. If we get an unrecognized event, then we don't want to set the data source state to valid. That would clear any existing error.

Reply:
I'm wondering if this shouldn't be a tri-bool and then we have a little different handling, where:

1. We only set it to true if we have processed a valid event.
2. Failure to decode an expected event fails would set this to false.
3. An unknown event type would be ignored, with no change to this value, as a sort of forward compatible guard against new event types we want to add.
4. We only execute line 294 if it is true, and avoid the false and new third (null?) state.

Reply:
Seems like a good idea. 

```
enum ProcessingState {
    EVENT_DECODED,
    EVENT_IGNORED,
    EVENT_MALFORMED
}
```
or similar.